src/server/telemetry.js

```diff
@@ -0,0 +1,168 @@
+  replace(http, 'request', networkRequestWrapper.bind(this), this.replacements, 'network');
+  replace(https, 'request', networkRequestWrapper.bind(this), this.replacements, 'network');
+};
+
+function networkRequestWrapper(orig) {
+  var telemeter = this.telemeter;
+  var self = this;
+
+  return function(url, options, cb) {
+    var mergedOptions = urlHelpers.mergeOptions(url, options, cb);
+
+    var metadata = {
+      method: mergedOptions.options.method || 'GET',
+      url: urlHelpers.constructUrl(mergedOptions.options),
+      status_code: null,
+      start_time_ms: _.now(),
+      end_time_ms: null
+    };
+
+    if (self.autoInstrument.networkRequestHeaders) {
+      metadata.request_headers = mergedOptions.options.headers;
+    }
+    telemeter.captureNetwork(metadata, 'http');
+
+    // Call the original method with the original arguments and wrapped callback.
+    var wrappedArgs = Array.from(arguments);
+    var wrappedCallback = responseCallbackWrapper(self.autoInstrument, metadata, mergedOptions.cb);
+    if (mergedOptions.cb) {
+      wrappedArgs.pop();
+    }
+    wrappedArgs.push(wrappedCallback);
+    var req = orig.apply(https, wrappedArgs);
+
+    req.on('error', err => {
+      metadata.status_code = 0;
+      metadata.error = [err.name, err.message].join(': ') ;
+    });
+
+    return req;
+  }
+}
+
+function responseCallbackWrapper(options, metadata, callback) {
+  return function (res) {
+    metadata.end_time_ms = _.now();
+    metadata.status_code = res.statusCode;
+    metadata.response = {};
+    if (options.networkResponseHeaders) {
+      metadata.response.headers = res.headers;
+    }
+
+    if (callback) {
+      return callback.apply(undefined, arguments);
+    }
+  }
+}
+
+Instrumenter.prototype.captureNetwork = function(metadata, subtype, rollbarUUID) {
+  return this.telemeter.captureNetwork(metadata, subtype, rollbarUUID);
+};
+
+Instrumenter.prototype.deinstrumentConsole = function() {
+  restore(this.replacements, 'log');
+};
+
+Instrumenter.prototype.instrumentConsole = function() {
+  var telemeter = this.telemeter;
+
+  var stdout = process.stdout;
+  replace(stdout, 'write', function(orig) {
+    return function(string) {
+      telemeter.captureLog(string, 'info');
+      return orig.apply(stdout, arguments);
+    }
+  }, this.replacements, 'log');
+
+  var stderr = process.stderr;
+  replace(stderr, 'write', function(orig) {
+    return function(string) {
+      telemeter.captureLog(string, 'error');
+      return orig.apply(stderr, arguments);
+    }
+  }, this.replacements, 'log');
+};
+
+function replace(obj, name, replacement, replacements, type) {
+  var orig = obj[name];
+  obj[name] = replacement(orig);
+  if (replacements) {
+    replacements[type].push([obj, name, orig]);
+  }
+}
+
+function restore(replacements, type) {
+  var b;
+  while (replacements[type].length) {
+    b = replacements[type].shift();
+    b[0][b[1]] = b[2];
+  }
+}
+
```

Comment:
Nit: Even though these are covered by integration tests later, I think unit tests for these would go a long way toward self-documenting how they work together and the intent of the implementation.

Reply:
I agree, but they can't be accessed as is without exporting them, and doing so could be considered an interface violation (depending on where one stands on that debate.)

These happen to be duplicated code from browser/telemetry.js and exctracting them into a shared module would eliminate the code duplication and create reasonable access for unit tests. I didn't deem that to be in scope for this PR though.

Reply:
Ok, then I think commenting this is duplicated code and an opportunity to refactor later with unit tests suffices to help future maintainers.

Reply:
Comment added. https://github.com/rollbar/rollbar.js/pull/943/commits/14a69df80c7c805b3cd1824be2a00b4081a17db4